### PR TITLE
LPS-197957 and LPS-197959 Persist page rules in backend

### PR DESCRIPTION
**Pull request by lfbesada**
https://liferay.atlassian.net/browse/LPS-197957
https://liferay.atlassian.net/browse/LPS-197959

cc: @sandrodw3 @victorg1991

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by victorg1991** [comment]
Divino el frontend 👏

**Comment by lfbesada** [comment]
@jkappler  thanks for pointing it out! :heart: 

@victorg1991 @sandrodw3 I think we should add a task to the story to handle errors in the backend part and prepare the front part to receive and display them. What do you think?

Regards

**Comment by jkappler** [comment]
I'm totally okay with it @lfbesada, as long as we handle them ❤️

**Comment by jkappler** [comment]
ci:forward

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/UpdateRuleMVCActionCommand.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+		ThemeDisplay themeDisplay = (ThemeDisplay)actionRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		return JSONUtil.put(
+			"layoutData",
+			LayoutStructureUtil.updateLayoutPageTemplateData(
+				themeDisplay.getScopeGroupId(),
+				ParamUtil.getLong(actionRequest, "segmentsExperienceId"),
+				themeDisplay.getPlid(),
+				layoutStructure -> layoutStructure.updateLayoutStructureRule(
```
[the remark this answers is not in the document]
Comment (sandrodw3) [reply modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/UpdateRuleMVCActionCommand.java:49]: Same 😇

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/AddRuleMVCActionCommand.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		throws Exception {
+
+		JSONObject jsonObject = _jsonFactory.createJSONObject();
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)actionRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		jsonObject.put(
+			"layoutData",
+			LayoutStructureUtil.updateLayoutPageTemplateData(
```
[the remark this answers is not in the document]
Comment (sandrodw3) [reply modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/AddRuleMVCActionCommand.java:49]: Cool, as accorded, we'll address it in another subtask 🥳

**Comment by sandrodw3** [comment]
Cool @jkappler @lfbesada! 😄 I'll create the subtask for it!

**Comment by sandrodw3** [comment]
ci:stop

**Comment by sandrodw3** [comment]
https://github.com/liferay-echo/liferay-portal/pull/14216

**Comment by jkappler** [comment]
ci:forward

**Comment by lfbesada** [comment]
ci:forward

**Comment by lfbesada** [comment]
ci:forward:force

**Pull request closed without merging**